### ISSUE TITLE
Pieces Desktop App - Lost Connection to PiecesOS

### ISSUE THREAD

**Issue by rafael-a-g-n** [pieces-app/support#531]
_PiecesOS Information_

**ID:** 18986c5c-5aef-40f4-ac00-93da40fd8b47
**User:** 1446bb64-5370-4644-a0ee-5ab3b4f3a7ac
**Current Version:** Unknown
**Min Version Supported:** 11.0.0
**Max Version Supported:** 12.0.0
**Local Port:** 127.0.0.1:39300

_Device Information_

**Operating System:** Windows
**Current Version:** "Windows 10 Pro" 10.0 (Build 26100)
**Min Version Supported:** Windows 10 Build 1903

Pieces for developers always loosing connection and self closing pieces OS

**Comment by mark-at-pieces** [comment]
Hey @rafael-a-g-n Thanks again for adding another issue, seems to bee you have experienced some funky behavior.

First, here is a link to my calendar if you want to chat about all the bugs you found in real time.(here is the [link](https://github.com/pieces-app/support/issues/530))

Second, if you would like to share a video that would be extremely helpful.

**Comment by tsavo-at-pieces** [comment]
Yep @laborer123 if you can let me know where your com.pieces.os folder is on you operating system that would be amazing. 

We'll get that edge case covered and make sure it is solved in tonight's patch!

**Comment by rafael-a-g-n** [comment]
Ok, I solved my problem of Pieces OS always disconnecting and restarting, if anyone else has this problem. Here's what I did. 
1 - uninstalled it (both OS and For Developers).
2 - went to my Documents(folder) and deleted the 2 folders there (com.pieces.os and com.pieces.pfd).
3 - reinstalled, and voila, Eureka!!!

missing step 2, and just uninstall and reinstall didn't work, same issue would occur time and time again.
Hope this helps someone on the same boat. After this all plugins problem's were also solved, at least for me.

Props to Pieces OS dev's for all the hard work putting this awesome "PIECE" of software for us all.

**Comment by tsavo-at-pieces** [comment]
Absolutely - also good news... we've got a patch rolling out tonight as well that will handle alternative installation locations gracefully! 

Really appreciate the support and help here... super excited for the upcoming releases. This one lays a ton of foundation for what's next.

Cheers, 
-Tsavo

**Comment by mack-at-pieces** [comment]
> Ok, I solved my problem of Pieces OS always disconnecting and restarting, if anyone else has this problem. Here's what I did. 1 - uninstalled it (both OS and For Developers). 2 - went to my Documents(folder) and deleted the 2 folders there (com.pieces.os and com.pieces.pfd). 3 - reinstalled, and voila, Eureka!!!
> 
> missing step 2, and just uninstall and reinstall didn't work, same issue would occur time and time again. Hope this helps someone on the same boat. After this all plugins problem's were also solved, at least for me.
> 
> Props to Pieces OS devs for all the hard work putting this awesome "PIECE" of software for us all.

This is really great to hear! Side note for others who may be seeing this:

Removing the `com.pieces.os` folder will also remove all of your saved code snippets and copilot chats. If you are OK with this, proceed. If not, feel free to connect with us and we can help repair the database if something is corrupted 👍🏼
